src/ripple/nodestore/impl/Shard.h

```diff
@@ -160,21 +162,30 @@ class Shard final
         utilized and the second item is the number of file descriptors required.
     */
     [[nodiscard]] std::pair<std::uint64_t, std::uint32_t>
-    getFileInfo() const;
+    getFileInfo() const noexcept;
 
     [[nodiscard]] State
-    getState() const
+    getState() const noexcept
     {
         return state_;
     }
 
+    /** Returns a value 0 - 1.0 signifying how complete the current state
+        of the shard is. This excludes the final state, where 0 is returned.
+     */
+    [[nodiscard]] float
+    getProgress() const noexcept
+    {
+        return progress_;
+    }
```

Comment:
This is where the `float` gets started.  I'll suggest that the `float` could readily be replaced by a `struct` like this:
```
    /** Returns a struct signifying how complete the current state
        of the shard is. This excludes the final state, where 0 is returned.
     */
    struct Progress {
        std::uint32_t const soFar;
        std::uint32_t const target;
    };
    [[nodiscard]] Progress
    getProgress() const noexcept
    {
        return {progress_, maxLedgers_};
    }
```
The `struct` provides names for the members, so it's easy to discern which is the numerator and which the denominator.  The `maxLedgers_` member is `const`, so it doesn't need to be atomic.  That allows `progress_` to be turned into an `atomic<std::uint32_t>` member, which simply records `numStored` wherever that's calculated.